src/org/etools/j1939_84/controllers/part11/Part11Step13Controller.java

```diff
@@ -53,33 +82,505 @@ public class Part11Step13Controller extends StepController {
+                                     }
+                                     if (spn.getSlot().toValue(spn.getBytes()) > 0) {
+                                         // 6.11.13.8.d - Fail each active 100 hr array value that is greater than zero
+                                         addFailure("6.11.13.8.d - Active 100 hr array value received is greater than zero from "
+                                                 + module.getModuleName() + " for " + spn);
+                                     }
+                                     // FIXME: the need to be implemented
+                                     // @Joe just need to implement these yet.
+                                     // 6.11.13.8.e - Fail if active 100 hrs engine hours < 600 seconds. (where
+                                     // supported)
+                                     // 6.11.13.8.f. Warn for all active 100 hr vehicle distance => 0.25 km. (Where
+                                     // supported).
+                                     // 6.11.13.8.g. Warn for active 100 hr EOE <= 0.5 kW-hr (where supported)
+                                 });
+            }
+        }
+    }
+
+    private void testSp12691(OBDModuleInformation module) {
+        // 6.11.13.9 Actions8 for all MY2022+ Engines
+        // 6.11.13.9.a - DS request message to ECU that indicated support in DM24 for upon request
+        // SP 12691 (GHG Tracking Lifetime Active Technology Index) for PG 64257 Green House Gas Lifetime Active
+        // Technology Tracking.
+        var ghgPackets = requestPackets(module.getSourceAddress(),
+                                        GHG_TRACKING_LIFETIME_GREEN_HOUSE_PG);
+
+        GenericPacket packetForPg = haveResponseWithPg(ghgPackets, GHG_TRACKING_LIFETIME_GREEN_HOUSE_PG);
+        if (packetForPg == null) {
+            // 6.11.13.10.a. Warn PG query where no response was received.
+            addWarning("6.11.13.10.a - No response was received from "
+                    + module.getModuleName() + " for PG "
+                    + GHG_TRACKING_LIFETIME_GREEN_HOUSE_PG);
+        } else {
+            packetForPg.getSpns()
+                       .forEach(spn -> {
+                           // 6.11.13.10.b. Fail any accumulator value received that is greater
+                           // than FAFFFFFFh.
+                           if (spn.getRawValue() > 0xFAFFFFFFL) {
+                               addFailure("6.11.13.10.b - Bin value received is greater than 0xFAFFFFFF(h) from "
+                                       + module.getModuleName() + " for " + spn);
+                           }
+                           // FIXME: this needs to implemented when the dataRepo is fixed
+                           // @Joe: just need to add a call and if adding the failure when the dataRepo bug is fixed
+                           // 6.11.13.10.c. Fail PG query where any index value received is
```

Comment:
This requirement doesn't depend upon the dataRepo.  You need to look at a specific SPN here.